packages/update/src/update.ts

```diff
@@ -170,17 +175,27 @@ export async function updateProject(dir: string) {
 
   // Update routes folder
   const project = new tsmorph.Project();
-  const sfs = project.addSourceFilesAtPaths(
-    path.join(dir, "**", "*.{js,jsx,ts,tsx}"),
-  );
 
-  // Filter out node_modules and vendor files for user display
-  const userFiles = sfs.filter((sf) => !HIDE_FILES.test(sf.getFilePath()));
+  // First pass: count files (without storing them) so we can size the progress
+  // bar while still skipping heavy dirs early.
+  let totalFiles = 0;
+  let userFileCount = 0;
+  for await (
+    const entry of walk(dir, {
+      includeDirs: false,
+      includeFiles: true,
+      exts: ["js", "jsx", "ts", "tsx"],
+      skip: [SKIP_DIRS],
+    })
+  ) {
+    totalFiles++;
+    if (!HIDE_FILES.test(entry.path)) userFileCount++;
+  }
```

Comment:
Let's store the found paths, so that we don't need to walk the file system again a second time.

Reply:
Agreed. I just pushed a new commit.